### PR TITLE
OC-12789: Fix scheduled extract errors

### PR DESCRIPTION
**Pull request by slclinica**
1. Added job name to the email error message
2. Removed delete archived dataset file functionality for scheduled jobs
3. Show failed when job fails instead of leave IN_PROGRESS
4. Use local HashMap variable to retrieve user roles instead of global variable thats causing confusion in variable type

### CODE REVIEW

File: src/main/java/core/org/akaza/openclinica/dao/login/UserAccountDAO.java [hunk 1]
```diff
@@ -123,17 +123,19 @@ public void setPrivilegeTypesExpected() {
 
     }
 
-    public void setRoleTypesExpected() {
+    public HashMap setRoleTypesExpected() {
         this.unsetTypeExpected();
```
Comment (svadla-oc) [inline src/main/java/core/org/akaza/openclinica/dao/login/UserAccountDAO.java:127]: you don't need this anymore.

File: src/main/java/core/org/akaza/openclinica/job/XsltTransformJob.java [hunk 2]
```diff
@@ -570,17 +570,28 @@ protected void executeInternal(JobExecutionContext context) {
             postErrorMessage(e.getMessage(), context);
             logger.error("Error executing extract", e);
             exceptions = true;
+        } catch (NullPointerException e) {
```
Comment (svadla-oc) [inline src/main/java/core/org/akaza/openclinica/job/XsltTransformJob.java:573]: why are you catching NPE? It's usually not a good idea to catch RuntimeExceptions

File: src/main/java/core/org/akaza/openclinica/dao/login/UserAccountDAO.java (hunk 1 above)
Comment (slclinica) [reply src/main/java/core/org/akaza/openclinica/dao/login/UserAccountDAO.java:127]: It's removed?

**Comment by svadla-oc** [comment]
👍

**Pull request merged**